### PR TITLE
[quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30262 [quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`**

Summary:
`get_qparams` returns all parameters that's needed to call quantize function

Test Plan:
python test/test_jit.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18645047](https://our.internmc.facebook.com/intern/diff/D18645047)

### CODE REVIEW

File: torch/csrc/jit/passes/quantization.cpp [hunk 1]
```diff
@@ -544,9 +544,9 @@ std::tuple<IValue, IValue> QuantizeHelper::getQParams(Value* v) {
       v->debugName(),
       " exists.");
   auto om = module_.attr(observer_name.value()).toModule();
-  auto calculate_qparams = om.get_method("calculate_qparams");
-  IValue qparams = calculate_qparams(std::vector<IValue>());
-  checkCalculateQParamsResult(qparams);
+  auto get_qparams = om.get_method("get_qparams");
```
Comment (raghuramank100) [inline torch/csrc/jit/passes/quantization.cpp]: NIT: Might be better to use a more descriptive name than 'om', perhaps: 'observer_module'

File: torch/csrc/jit/passes/quantization.cpp [hunk 2]
```diff
@@ -503,16 +503,16 @@ Value* QuantizeHelper::removeObserver(
...
       qparams.isTuple(),
-      "`calculate_qparams` function is expected to return a "
+      "`get_qparams` function is expected to return a "
       "Tuple, but got:",
       qparams.tagKind());
   auto tp = qparams.toTuple();
   TORCH_CHECK(
       tp->elements().size() == 2 || tp->elements().size() == 3,
-      "`calculate_qparams` function is expected to reutrn a "
+      "`get_qparams` function is expected to reutrn a "
```
Comment (raghuramank100) [inline torch/csrc/jit/passes/quantization.cpp]: NIT: typo 'return'

File: torch/nn/quantized/modules/utils.py [hunk 3]
```diff
@@ -1,13 +1,13 @@
 import torch
 
 def _quantize_weight(float_wt, observer):
+    wt_scale, wt_zp = observer.calculate_qparams()
```
Comment (raghuramank100) [inline torch/nn/quantized/modules/utils.py]: Would it be cleaner to do:
```python
q_params = observer.get_qparams()
if ....
    qweight = torch.quantize_per_tensor(float_wt, float(q_params[0]), int(q_params[1]), torch.qint8)
elif ...
    qweight = torch.quantize_per_tensor(float_wt, q_params[0].to(torch.double), q_params[1].to(torch.int64), q_params[2], torch.qint8)
```

**Review by raghuramank100 — APPROVED** [review]
Looks good, have a few suggestions.

File: torch/nn/quantized/modules/utils.py (hunk 3 above)
Comment (jerryzh168) [reply torch/nn/quantized/modules/utils.py]: probably better to keep this way... I think maybe we don't need get_qparams, we can get access to ch_axis through observer module

**Comment by jerryzh168** [comment]
I'll remove `get_qparams` in another PR, we can refactor the code of getting qparams from observer modules later

**Pull request closed without merging**